test/integration/package_id/compatible_test.py

```diff
@@ -706,3 +706,78 @@ def validate_build(self):
             pkga = liba["packages"][0][pkg_index]
             assert pkga["info"]["compatibility_delta"] == {"settings": [["compiler.cppstd", "14"]]}
             assert pkga["build_args"] == "--requires=liba/0.1 --build=compatible:liba/0.1"
+
+
+def test_compatibility_new_setting_forwards_compat():
+    """ This test tries to reflect the following scenario:
+    - User adds a new setting (libc.version in this case)
+    - This setting is forward compatible
+    How is it solved with compatibility.py? Like this:
+    """
+    settings_user_file = "libc_version: [1, 2, 3]"
+    tc = TestClient()
+    tc.save_home({"settings_user.yml": settings_user_file})
+    tc.save({"conanfile.py": GenConanfile("dep", "1.0")
+            .with_settings("libc_version", "compiler")})
+    tc.run("create . -s=libc_version=2 -s=compiler.cppstd=17")
+    dep_package_id = tc.created_package_id("dep/1.0")
+    tc.run("install --requires=dep/1.0 -s=libc_version=3 -s=compiler.cppstd=17", assert_error=True)
+    # We can't compile, because the dep is not compatible
+    assert "Missing prebuilt package for 'dep/1.0'" in tc.out
+
+    # Let's create a compatibility extensions
+    libc_compat = textwrap.dedent("""
+    from conan.tools.scm import Version
+
+    def libc_compat(conanfile):
+        # Do we have the setting?
+        libc_version = conanfile.settings.get_safe("libc_version")
+        if libc_version is None:
+            return []
+        available_libc_versions = conanfile.settings.libc_version.possible_values()
+        ret = []
+        for possible_libc_version in available_libc_versions:
+            if Version(possible_libc_version) <= Version(libc_version):
+                ret.append({"settings": [("libc_version", possible_libc_version)]})
+        return ret
+    """)
+
+    # And add it to the compatibility.py plugin
+    compatibility_plugin = textwrap.dedent("""
+    from cppstd_compat import cppstd_compat
+    from libc_compat import libc_compat
+
+    def compatibility(conanfile):
+        configs = cppstd_compat(conanfile)
+        libc_configs = libc_compat(conanfile)
+        results = []
+        # First only the cppstd settings
+        for config in configs:
+            results.append(config)
+        # Now the libc settings by themselves
+        for libc_config in libc_configs:
+            results.append(libc_config)
+        # And now the combination of both
+        for config in configs:
```

Comment:
It is not addition. I am afraid you need to do the combination, see the current implementation of cppstd:
```python
for factor in factors:
        if not combinations:
            combinations = factor
            continue
        new_combinations = []
        for comb in combinations:
            for f in factor:
                new_comb = comb.copy()
                new_comb.update(f)
                new_combinations.append(new_comb)
        combinations.extend(new_combinations)
```